src/services/contactService.js

```diff
@@ -1,6 +1,21 @@
 const Contact = require('../models/contactModel')
 
 // get all with optional pagination
+exports.list = async (limit = 5, page = 1, name = null) => {
+  let query = {
+    limit: parseInt(limit < 1 ? 1 : limit), //set minimum limit to 1
+    skip: limit * (page - 1) //skip for pagination
```

Comment:
- check if parsing should happen in service layer
- if yes: parse both **limit** and **skip**

Reply:
parse removed from service. validation added to controller